src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackProcessor.kt

```diff
@@ -48,6 +53,33 @@ class CallbackProcessor {
     }
 
     private fun handleAttachRecord(caseDetails: CaseDetails): List<String> {
-        return listOf()
+        return if (caseDetails.data != null) {
+            startAttachEvent(caseDetails)
+        } else {
+            logger.error { "No case details supplied eventId: ${caseDetails.id}" }
+            listOf("Internal Error: no case data")
+        }
+    }
+
+    private fun startAttachEvent(caseDetails: CaseDetails): List<String> {
+        try {
+            val authenticator = authenticatorFactory.createForJurisdiction(caseDetails.jurisdiction)
+            ccdApi.startEventForCaseWorker(
+                authenticator.userToken,
+                authenticator.serviceToken,
+                authenticator.userDetails.id,
+                caseDetails.jurisdiction,
+                caseDetails.caseTypeId,
+                caseDetails.data["attachToCaseReference"] as String?,
+                "TBD"
+            )
+            return listOf()
+        } catch (e: FeignException) {
+            logger.error(e) {}
+            return listOf("Internal Error: response ${e.status()} submitting event")
+        } catch (e: Exception) {
+            logger.error(e) {}
+            return listOf("Internal Error: ${e::class.simpleName}:${e.message}")
```

Comment:
probably should just be Internal server error, if you're feeling brave finding out the request id for app insights and returning that here too?